### ISSUE TITLE
Building COLMAP on Mac

### ISSUE THREAD

**Issue by apurvazaveri** [colmap/colmap#1667]
**Describe the bug**
Building Colmap on Mac produces the following the linker error during "make".

**To Reproduce**
git clone https://github.com/colmap/colmap.git
cd colmap
# on osx/colmap
mkdir build
cd build
cmake .. -DQt5_DIR=/usr/local/opt/qt5/lib/cmake/Qt5
make 
**Expected behavior**

**Screenshots**
[  2%] Built target flann
[  2%] Built target lsd
[  4%] Built target pba
[  9%] Built target poisson_recon
[ 14%] Built target sift_gpu
[ 16%] Built target sqlite3
[ 35%] Built target vlfeat
[ 95%] Built target colmap
[ 95%] Linking CXX executable colmap
ld: warning: dylib (/usr/local/lib/libglog.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libceres.2.1.0.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libglog.0.6.0.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libomp.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libgmpxx.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libmpfr.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: dylib (/usr/local/lib/libgmp.dylib) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_filesystem-mt.a(path.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(value_semantic.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(options_description.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_filesystem-mt.a(directory.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_filesystem-mt.a(operations.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_filesystem-mt.a(utf8_codecvt_facet.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(convert.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(variables_map.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(parsers.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(cmdline.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_filesystem-mt.a(exception.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(utf8_codecvt_facet.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(config_file.o)) was built for newer macOS version (12.0) than being linked (11.3)
ld: warning: object file (/usr/local/lib/libboost_program_options-mt.a(positional_options.o)) was built for newer macOS version (12.0) than being linked (11.3)
Undefined symbols for architecture x86_64:
  "boost::filesystem::detail::directory_iterator_construct(boost::filesystem::directory_iterator&, boost::filesystem::path const&, unsigned int, boost::system::error_code*)", referenced from:
      colmap::GetFileList(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libcolmap.a(misc.cc.o)
      colmap::GetDirList(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libcolmap.a(misc.cc.o)
  "std::__1::basic_filebuf<char, std::__1::char_traits<char> >::basic_filebuf()", referenced from:
      boost::program_options::basic_parsed_options<char> boost::program_options::parse_config_file<char>(char const*, boost::program_options::options_description const&, bool) in libboost_program_options-mt.a(parsers.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/exe/colmap] Error 1
make[1]: *** [src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
make: *** [all] Error 2

**Desktop (please complete the following information):**
 - OS: Mac OS 12.6 Monterey

**Comment by ahojnnes** [comment]
Seems like you were using homebrew to install dependencies? I have seen similar behavior in the past where homebrew downloaded pre-built binaries from its cache, but these were built for a different Mac version. Might be worth reinstalling homebrew and trying again. Maybe force homebrew to compile all binaries from your machine instead of downloading the pre-built ones. Unfortunately, I cannot use with detailed instructions. I am not using Mac these days anymore

**Comment by apurvazaveri** [comment]
Yes, I am using homebrew to install dependencies. I tried the following so far with no luck:

brew doctor
brew reinstall boost
